src/catchup/ApplyBucketsWork.cpp

```diff
@@ -193,8 +226,22 @@ ApplyBucketsWork::onRun()
         return State::WORK_RUNNING;
     }
 
-    CLOG(DEBUG, "History") << "ApplyBuckets : done, restarting merges";
-    mApp.getBucketManager().assumeState(mApplyState, mMaxProtocolVersion);
+    CLOG(INFO, "History") << "ApplyBuckets : done, restarting merges";
+    // Use old-style merges from HAS
+    bool useHASFutureBuckets =
+        mMaxProtocolVersion < Bucket::FIRST_PROTOCOL_SHADOWS_REMOVED;
+    mApp.getBucketManager().assumeState(mApplyState, mMaxProtocolVersion,
+                                        useHASFutureBuckets);
+
+    // To avoid tripping off the network during online catchup post protocol 12,
```

Comment:
Not entirely clear I understand this comment. Could you expand it a bit? What does "tripping off the network" mean here, and what is a promotion ledger?

Reply:
When buckets are applied, we won't have ready outputs before joining the network. This means that if we mark catchup as "successful", and start closing ledgers, we might block for a while, waiting for a merge to be resolved, so that the future bucket can be promoted into the bucketlist at that ledger. If resolution takes long enough, a node might lose sync. 

I'll reword this comment to avoid confusion.

Reply:
Great, clearer now, thanks!